kiegroup/kie-benchmarks
# BAPL-2265: Test case to verify task operation with a large number of active process instances using timers

The idea of this test case is to run it twice, first again an empty database to collect baseline values, then against a database with a large number of active process instances using timers. The test class can be used to generate these instances when run with *-Dtimertest.completetask=false*

**Prerequisites**:
- Run kie-server configured with database backed EJB timers 
- Create user perfUser:perfUser1234 

**Test steps**:
- Build the complete kie-benchmarks project
- cd jbpm-benchmarks/kieserver-performance-tests-jmh
- Run the test against a clean DB: 
```
java -jar target/kieserver-benchmarks.jar -bm avgt -i 10 -t 100 -p remoteAPI=REST -jvmArgsAppend '-Dkieserver.host="localhost" -Dkieserver.username="perfUser" -Dkieserver.password="perfUser1234" -Dkieserver.port=8080 -Dkieserver.name="kie-server" -Ddatasource.jndi=java:jboss/pg_rhpam7 -DremoteAPI=REST -Dtimertest.completetask=true' -rff results.csv -e LServer*
```
- Record the test results
- Recreate the database
- Populate the database with a significant number of active instances:
```
java -jar target/kieserver-benchmarks.jar -bm avgt -i 50 -t 100 -p remoteAPI=REST -jvmArgsAppend '-Dkieserver.host="localhost" -Dkieserver.username="perfUser" -Dkieserver.password="perfUser1234" -Dkieserver.port=8080 -Dkieserver.name="kie-server" -Ddatasource.jndi=java:jboss/pg_rhpam7 -DremoteAPI=REST -Dtimertest.completetask=false' -rff results.csv 
```
(note the higher number of iterations and the *-Dtimertest.completetask=false* setting)
- Check the DB for a significant number of entries in ProcessInstanceInfo, jboss_ejb_timer
- Repeat the test run as before:
```
java -jar target/kieserver-benchmarks.jar -bm avgt -i 10 -t 100 -p remoteAPI=REST -jvmArgsAppend '-Dkieserver.host="localhost" -Dkieserver.username="perfUser" -Dkieserver.password="perfUser1234" -Dkieserver.port=8080 -Dkieserver.name="kie-server" -Ddatasource.jndi=java:jboss/pg_rhpam7 -DremoteAPI=REST -Dtimertest.completetask=true' -rff results.csv
```
